frontend/src/lib/StatusUtils.ts

```diff
@@ -40,11 +40,11 @@ export enum NodePhase {
 export function hasFinished(status?: NodePhase): boolean {
   switch (status) {
     case NodePhase.SUCCEEDED: // Fall through
+      return true;
     case NodePhase.FAILED: // Fall through
     case NodePhase.ERROR: // Fall through
     case NodePhase.SKIPPED: // Fall through
-    case NodePhase.TERMINATED:
-      return true;
+    case NodePhase.TERMINATED: // Fall through
     case NodePhase.PENDING: // Fall through
     case NodePhase.RUNNING: // Fall through
     case NodePhase.TERMINATING: // Fall through
```

Comment:
No, @IronPan this is problematic with the existing use of `hasFinished`. This function is supposed to indicate that the the run had stopped running for any reason. For example, it is used in the `RunDetails` page to determine whether to continue autorefreshing because a `pending` run might update over time, but a `failed` run never will.

Is there any reason you can't create a separate function called `hasSucceeded` or something like that?

Reply:
The goal for this change is to continue to refresh on non succeeded runs because those might still change the status once user click retry.